File: android/src/main/java/expo/modules/bbplayer/ExpoBBPlayerModule.kt
package expo.modules.bbplayer

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoBBPlayerModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoBBPlayer")

    View(ExpoBBPlayerView::class) {
      Prop("jsonUrl") { view: ExpoBBPlayerView, url: String ->
        view.setJsonUrl(url)
      }

      Prop("options") { view: ExpoBBPlayerView, options: HashMap<String, Any?> ->
        view.setOptions(options)
      }

      OnViewDidUpdateProps {view ->
        view.setupPlayer()
      }

      AsyncFunction("adMediaHeight") { view: ExpoBBPlayerView ->
        view.adMediaHeight()
      }

      AsyncFunction("adMediaWidth") { view: ExpoBBPlayerView ->
        view.adMediaWidth()
      }

      AsyncFunction("adMediaClip") { view: ExpoBBPlayerView ->
        view.adMediaClip()
      }

      AsyncFunction("controls") { view: ExpoBBPlayerView ->
        view.controls()
      }

      AsyncFunction("duration") { view: ExpoBBPlayerView ->
        view.duration()
      }

      AsyncFunction("inView") { view: ExpoBBPlayerView ->
        view.inView()
      }

      AsyncFunction("mode") { view: ExpoBBPlayerView ->
        view.mode()
      }

      AsyncFunction("muted") { view: ExpoBBPlayerView ->
        view.muted()
      }

      AsyncFunction("phase") { view: ExpoBBPlayerView ->
        view.phase()
      }

      AsyncFunction("playoutData") { view: ExpoBBPlayerView ->
        view.playoutData()
      }

      AsyncFunction("projectData") { view: ExpoBBPlayerView ->
        view.projectData()
      }

      // using playerState as name to avoid collision with internal the state function
      AsyncFunction("playerState") { view: ExpoBBPlayerView ->
        view.state()
      }

      AsyncFunction("volume") { view: ExpoBBPlayerView ->
        view.volume()
      }

      AsyncFunction("autoPlayNextCancel") { view: ExpoBBPlayerView ->
        view.autoPlayNextCancel()
      }

      AsyncFunction("collapse") { view: ExpoBBPlayerView ->
        view.collapse()
      }

      AsyncFunction("expand") { view: ExpoBBPlayerView ->
        view.expand()
      }

      AsyncFunction("enterFullscreen") { view: ExpoBBPlayerView ->
        view.enterFullscreen()
      }

      AsyncFunction("exitFullscreen") { view: ExpoBBPlayerView ->
        view.exitFullscreen()
      }

      AsyncFunction("pause") { view: ExpoBBPlayerView ->
        view.pause()
      }

      AsyncFunction("play") { view: ExpoBBPlayerView ->
        view.play()
      }

      AsyncFunction("seek") { view: ExpoBBPlayerView, offsetInSeconds: Int ->
        view.seek(offsetInSeconds)
      }

      AsyncFunction("setMuted") { view: ExpoBBPlayerView, muted: Boolean ->
        view.setMuted(muted)
      }

      AsyncFunction("setVolume") { view: ExpoBBPlayerView, volume: Double ->
        view.setVolume(volume)
      }

      Events(
        "onDidFailWithError",
        "onDidRequestCollapse",
        "onDidRequestExpand",
        "onDidTriggerCanPlay",
        "onDidRequestOpenUrl",
        "onDidSetupWithJsonUrl",
        "onDidTriggerAdError",
        "onDidTriggerAdFinished",
        "onDidTriggerAdLoaded",
        "onDidTriggerAdLoadStart",
        "onDidTriggerAdNotFound",
        "onDidTriggerAdQuartile1",
        "onDidTriggerAdQuartile2",
        "onDidTriggerAdQuartile3",
        "onDidTriggerAdStarted",
        "onDidTriggerAllAdsCompleted",
        "onDidTriggerAutoPause",
        "onDidTriggerAutoPausePlay",
        "onDidTriggerCanPlay",
        "onDidTriggerCustomStatistics",
        "onDidTriggerDurationChange",
        "onDidTriggerEnded",
        "onDidTriggerFullscreen",
        "onDidTriggerMediaClipFailed",
        "onDidTriggerMediaClipLoaded",
        "onDidTriggerModeChange",
        "onDidTriggerPause",
        "onDidTriggerPhaseChange",
        "onDidTriggerPlay",
        "onDidTriggerPlaying",
        "onDidTriggerProjectLoaded",
        "onDidTriggerRetractFullscreen",
        "onDidTriggerSeeked",
        "onDidTriggerSeeking",
        "onDidTriggerStall",
        "onDidTriggerStateChange",
        "onDidTriggerViewFinished",
        "onDidTriggerViewStarted",
        "onDidTriggerVolumeChange",
      )
    }
  }
}


   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   
   

File: ios/ExpoBBPlayerModule.swift
import ExpoModulesCore

public class ExpoBBPlayerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoBBPlayer")

    View(ExpoBBPlayerView.self) {
      Prop("jsonUrl") { (view, url: String) in
        view.setJsonUrl(url)
      }

      Prop("options") { (view, options: [String: Any]) in
        view.setOptions(options)
      }

      OnViewDidUpdateProps { view in
        view.setupPlayer()
      }

      AsyncFunction("adMediaHeight") { (view: ExpoBBPlayerView) in
        return view.adMediaHeight()
      }

      AsyncFunction("adMediaWidth") { (view: ExpoBBPlayerView) in
        return view.adMediaWidth()
      }

      AsyncFunction("adMediaClip") { (view: ExpoBBPlayerView) in
        return view.adMediaClip()
      }

      AsyncFunction("controls") { (view: ExpoBBPlayerView) in
        return view.controls()
      }

      AsyncFunction("duration") { (view: ExpoBBPlayerView) in
        return view.duration()
      }

      AsyncFunction("inView") { (view: ExpoBBPlayerView) in
        return view.inView()
      }

      AsyncFunction("mode") { (view: ExpoBBPlayerView) in
        return view.mode()
      }

      AsyncFunction("muted") { (view: ExpoBBPlayerView) in
        return view.muted()
      }

      AsyncFunction("phase") { (view: ExpoBBPlayerView) in
        return view.phase()
      }

      AsyncFunction("playoutData") { (view: ExpoBBPlayerView) in
        return view.playoutData()
      }

      AsyncFunction("projectData") { (view: ExpoBBPlayerView) in
        return view.projectData()
      }

      AsyncFunction("playerState") { (view: ExpoBBPlayerView) in
        return view.state()
      }

      AsyncFunction("volume") { (view: ExpoBBPlayerView) in
        return view.volume()
      }

      AsyncFunction("autoPlayNextCancel") { (view: ExpoBBPlayerView) in
        return view.autoPlayNextCancel()
      }

      AsyncFunction("collapse") { (view: ExpoBBPlayerView) in
        return view.collapse()
      }

      AsyncFunction("expand") { (view: ExpoBBPlayerView) in
        return view.expand()
      }

      AsyncFunction("enterFullscreen") { (view: ExpoBBPlayerView) in
        return view.enterFullscreen()
      }

      AsyncFunction("exitFullscreen") { (view: ExpoBBPlayerView) in
        return view.exitFullscreen()
      }

      AsyncFunction("pause") { (view: ExpoBBPlayerView) in
        return view.pause()
      }

      AsyncFunction("play") { (view: ExpoBBPlayerView) in
        return view.play()
      }

      AsyncFunction("seek") { (view: ExpoBBPlayerView, offsetInSeconds: Int) in
        return view.seek(offsetInSeconds)
      }

      AsyncFunction("setMuted") { (view: ExpoBBPlayerView, muted: Bool) in
        return view.setMuted(muted)
      }

      AsyncFunction("setVolume") { (view: ExpoBBPlayerView, volume: Double) in
        return view.setVolume(volume)
      }

      Events([
        "onDidFailWithError",
        "onDidRequestCollapse",
        "onDidRequestExpand",
        "onDidRequestOpenUrl",
        "onDidSetupWithJsonUrl",
        "onDidTriggerAdError",
        "onDidTriggerAdFinished",
        "onDidTriggerAdLoaded",
        "onDidTriggerAdLoadStart",
        "onDidTriggerAdNotFound",
        "onDidTriggerAdQuartile1",
        "onDidTriggerAdQuartile2",
        "onDidTriggerAdQuartile3",
        "onDidTriggerAdStarted",
        "onDidTriggerAllAdsCompleted",
        "onDidTriggerAutoPause",
        "onDidTriggerAutoPausePlay",
        "onDidTriggerCanPlay",
        "onDidTriggerCustomStatistics",
        "onDidTriggerDurationChange",
        "onDidTriggerEnded",
        "onDidTriggerFullscreen",
        "onDidTriggerMediaClipFailed",
        "onDidTriggerMediaClipLoaded",
        "onDidTriggerModeChange",
        "onDidTriggerPause",
        "onDidTriggerPhaseChange",
        "onDidTriggerPlay",
        "onDidTriggerPlaying",
        "onDidTriggerProjectLoaded",
        "onDidTriggerRetractFullscreen",
        "onDidTriggerSeeked",
        "onDidTriggerSeeking",
        "onDidTriggerStall",
        "onDidTriggerStateChange",
        "onDidTriggerViewFinished",
        "onDidTriggerViewStarted",
        "onDidTriggerVolumeChange",
      ])
    }
  }
}
